docs/index.md

```diff
@@ -1,3 +1,134 @@
+<!-- vim: set textwidth=110: -->
+
 # rocd
 
-Welcome to rocd documentation!
+We start from a small preview of a system as a whole to understand what is the role of `rocd` component. The
+purpose of the [Roc Toolkit](https://roc-streaming.org/toolkit/docs/about_project/overview.html) (further
+referred to as `roc`) is to implement real-time audio streaming over network.
+
+## Entities
+
+Here are the top-level components:
+
+**host**
+:   Computer where the `roc` instance is currently running on. Computer platform may be on of:
+
+    - PC (personal computer, usually a desktop device with monitor or a similar output device),
+    - Android,
+    - iOS (future plans),
+    - embedded (is actually the same as PC, but usually doesn't have a monitor).
+
+**rocd**
+:   Server component used for stream management and auto discovery over network. Contains a library and a
+    binary that wraps it in a daemon. A stand-alone library is needed for mobile versions with their specific
+    platform limitations (programming language, permission models, ...).
+
+**roc-cast**
+:   Component used for two primary purposes:
+
+    - Provide UI for user interaction with rocd.
+    - On Android (and — possibly — iOS) devices it implements a wrapper application with a so-called main
+      activity context to allow interaction between non-Java libraries (C/C++/...) with mobile sound devices.
```

Comment:
Don't quite get this section. Any android app has main activity. There is no wrapper app - roc-cast is regular android app; one of its jobs is to acquire permissions and pass context with granted permissions to rocd. It will use rocd via wrapping java *library*, which will be *part of rocd*.

Reply:
just rewrite everything related to Android by yourself please :)